ArduPlane/control_modes.cpp

```diff
@@ -149,13 +149,27 @@ void Plane::read_control_switch()
 uint8_t Plane::readSwitch(void) const
 {
     uint16_t pulsewidth = RC_Channels::get_radio_in(g.flight_mode_channel - 1);
+#if FLIGHT_MODE_COUNT <= 6
+    // Use the legacy ranges, for backwards compatibility.
     if (pulsewidth <= 900 || pulsewidth >= 2200) return 255;            // This is an error condition
     if (pulsewidth <= 1230) return 0;
     if (pulsewidth <= 1360) return 1;
     if (pulsewidth <= 1490) return 2;
     if (pulsewidth <= 1620) return 3;
     if (pulsewidth <= 1749) return 4;              // Software Manual
     return 5;                                                           // Hardware Manual
+#else
+    // All bets are off.
+
+    uint8_t range_per_mode = 1000 / FLIGHT_MODE_COUNT;
```

Comment:
must be missing something....how can the range per mode be 1000us? maybe100us?

Reply:
Hmm, what do you mean? It divides the full range by the number of modes to arrive at the range per mode. E.g. for 9 modes its 133us per mode.

Reply:
sorry...read the / as // a comment and ignored....looks right...sorry